tiledb/sm/query/reader_base.cc

```diff
@@ -1245,8 +1271,14 @@ Status ReaderBase::unfilter_tiles(
           // Skip non-existent attributes/dimensions (e.g. coords in the
           // dense case).
           if (tile_tuple == nullptr ||
-              std::get<0>(*tile_tuple).filtered_buffer().size() == 0)
+              std::get<0>(*tile_tuple).filtered_buffer().size() == 0) {
             return Status::Ok();
+          }
+
+          // If the fragment doesn't include timestamps
+          if (name == constants::timestamps && !fragment->has_timestamps()) {
```

Comment:
I see this check repeating many times, let's make it a function? Perhaps a method of fragment.